server/vitals.js

```diff
@@ -180,7 +182,14 @@ async function handleHeartbeat(req, res) {
     if (validationErrors.isEmpty()) {
       const coreId = req.body.coreid
       const location = await db.getLocationFromParticleCoreID(coreId)
```

Comment:
There is a speed optimization possible related to this line here. In general, calls to the database are slow. What do you think you could do here to make the `handleHeartbeat` function as lean as it can be.